lib/src/main/java/com/auth0/android/lock/Configuration.java

```diff
@@ -97,19 +90,17 @@ public Configuration(Application application, Options options) {
         String defaultDatabaseName = options.getDefaultDatabaseConnection();
         Set<String> connectionSet = connections != null ? new HashSet<>(connections) : new HashSet<String>();
         this.extraSignUpFields = options.getCustomFields();
-        this.defaultDatabaseConnection = filterDatabaseConnections(application.getDatabaseStrategy(), connectionSet, defaultDatabaseName);
-        this.enterpriseStrategies = filterStrategiesByConnections(application.getEnterpriseStrategies(), connectionSet);
-        this.passwordlessStrategies = filterStrategiesByConnections(application.getPasswordlessStrategies(), connectionSet);
-        this.activeDirectoryStrategy = filterStrategy(application.strategyForName(Strategies.ActiveDirectory.getName()), connectionSet);
-        this.defaultActiveDirectoryConnection = filteredDefaultADConnection(this.activeDirectoryStrategy);
-        this.socialStrategies = filterSocialStrategies(application.getSocialStrategies(), connectionSet);
+        this.defaultDatabaseConnection = filterDatabaseConnections(application.getConnections(), connectionSet, defaultDatabaseName);
+        this.enterpriseConnections = filterConnections(application.getConnections(), connectionSet, AuthType.ENTERPRISE);
+        this.passwordlessConnections = filterConnections(application.getConnections(), connectionSet, AuthType.PASSWORDLESS);
+        this.socialConnections = filterConnections(application.getConnections(), connectionSet, AuthType.SOCIAL);
         this.application = application;
         parseLocalOptions(options);
         boolean atLeastOneSocialModeEnabled = allowLogIn || allowSignUp;
         boolean atLeastOneDatabaseModeEnabled = atLeastOneSocialModeEnabled || allowForgotPassword;
-        this.classicLockAvailable = (!socialStrategies.isEmpty() && atLeastOneSocialModeEnabled) || (!enterpriseStrategies.isEmpty() && allowLogIn)
+        this.classicLockAvailable = (!socialConnections.isEmpty() && atLeastOneSocialModeEnabled) || (!enterpriseConnections.isEmpty() && allowLogIn)
```

Comment:
@hzalaz This values are queried before showing the Lock screen to determine not only if the connections are available (after parsing the response from the cdn) but to also check if the user has disabled some screens. i.e.: 
* If `social connections` are available but both `sign up / log in screen` are disabled, it won't show it.
* If `enterprise connections` are available but `log in screen` is disabled, it won't show it.

I can do this same boolean logic on the view side if that's what you want.